Inferences/BinaryResolution.cpp

```diff
@@ -241,8 +227,21 @@ Clause* BinaryResolution::generateClause(Clause* queryCl, Literal* queryLit, Cla
 
   inf_destroyer.disable(); // ownership passed to the the clause below
   Clause *cl = Clause::fromStack(*resLits, inf);
-  if(env.options->proofExtra() == Options::ProofExtra::FULL)
+  if (env.options->questionAnswering() == Options::QuestionAnsweringMode::SYNTHESIS) {
+    Literal *qAnsLit = queryCl->getAnswerLiteral(), *rAnsLit = resultCl->getAnswerLiteral();
+    Literal* sqAnsLit = qAnsLit ? subs->applyToQuery(qAnsLit) : nullptr;
```

Comment:
Maybe I misunderstood but as far as I can tell here the answer literals can be null, but around `AnswerLiteralProcessors.cpp:167` you check that they shouldn't be null.

More generally, wouldn't it make more sense to completely skip creating the proof extra object if there is only one answer literal inherited?

Reply:
Re the first point: that is okay, we only get to `AnswerLiteralProcessors.cpp:167` if the clause has two answer literals.

Re the second point: yes, thanks for the suggestion!